R/messages.R

```diff
@@ -4,58 +4,176 @@
+    last_char <- substr(server, (nchar(server)+1)-1, nchar(server))
+    if (!is.na(last_char) && last_char != "/") {
+      server <- paste0(server, "/")
     }
+  }
 
+  url <- paste0(server, "__api__/server_settings")
+  server_settings <- NULL
+  result <- tryCatch (
+    {
+      "!DEBUG Sending GET request to `url`"
+      response <- httr::GET(
+        url,
+        httr::add_headers(Authorization = paste0("Key ", api_key)),
+        httr::write_memory()
+      )
+      list(success=TRUE, response=response)
+    },
+    error = function(err) {
+      "!DEBUG GET response threw an exception: `err`"
+
+      return (list(
+        success=FALSE, 
+        message=paste0("### API request to ", server, " has failed with error:",
+            "\n", err, 
+            "\n",
+            "\nPossible Solutions:",
+            "\n",
+            "\n-    If you have specified an API_KEY, confirm it is valid.",
+            "\n-    Confirm there is connectivity between the server itself and the address assigned to it: ", server, ".", 
+            "\n-    If using HTTPS along with self-signed certificates, you may need to allow the plumbertableau package to use HTTP instead, ", 
+            "by setting the environment variable *PLUMBERTABLEAU_USE_HTTP* to *TRUE* within the RStudio Connect application settings."
+        )
+      ))
+    }
+  )
+  if (!result$success) {
+    "!DEBUG Detected that an error has been returned from exception: `result`"
 
-    # Send warning message to the console if any of the above are TRUE
-    if (!rlang::is_null(message_contents)) {
-      message_contents <- paste(
+    message_contents <- paste0(
+      message_contents,
+      result$message,
+      sep = "\n\n---\n\n"
+    )
+    message_count <- message_count + 1
+  } else {
+    "!DEBUG GET response has returned `httr::http_status(result$response)$category`, `httr::http_status(result$response)$reason`, `httr::http_status(result$response)$message`"
+    
+    if (httr::http_error(result$response)) {
+      message_contents <- paste0(
         message_contents,
-        "\n\n#### Please reach out to your RStudio Connect administrator to fix these issues.",
-        sep = "\n"
+        paste0("### API request to ", server, " failed. ",
+        "\n-    Response: ", httr::http_status(result$response)$reason, ", ", httr::http_status(result$response)$message,
+        "\n",
+        "\nPossible Solution:",
+        "\n",
+        "\nDiagnose connectivity or access issue."
+        ),
+        sep = "\n\n---\n\n"
       )
+      message_count <- message_count + 1
 
-      rlang::warn(stringi::stri_replace_all(message_contents, regex = "\\* ?|#+ ", replacement = ""), .frequency = "once", .frequency_id = "rsc_warning")
+    } else {
+      server_settings <- httr::content(result$response, as="parsed")
+      "!DEBUG GET response was successful. Server settings = `server_settings`"
     }
   }
 
+  if (message_count > 0) {
+    return (message_contents)
+  }
```

Comment:
The ordering here seems off to me. It feels like we check for a bunch of things before we check if Tableau extensions are even supported on this RSC instance. Since we're returning here, we won't ever check for RSC compatibility unless all other checks pass.

Reply:
Talked through this and will leave it as is.